### ISSUE TITLE
User Story is missing in Timeline

### ISSUE THREAD

**Issue by shinglyu** [bzdeck/bzdeck#280]
Perhaps I'm missing something. But I can't find the "User Story" section in the timeline. [[example](https://bugzilla.mozilla.org/show_bug.cgi?id=1094759&redirect=no)]

**Comment by kyoshino** [comment]
Yeah I know. User Story is part of the flags so we can show that.

Votes, Crash Signature and perhaps some other fields are not available the API, so there's no way to show them for now.

**Comment by shinglyu** [comment]
(Y)

**Comment by shinglyu** [comment]
This doesn't look complicated. Can I take this?

**Comment by kyoshino** [comment]
Sure!

**Comment by shinglyu** [comment]
I an not sure how to turn the `\n` in the user story text into `<br>`. I tried something like

```
_bug.cf_user_story = this.bug.cf_user_story.replace(/\n{2,}/gm, '').replace(/\n/gm, '<br>');
```

But it get rendered as text, not a linebreak.

**Comment by shinglyu** [comment]
Ah ok, so it seems like flaretail.js changed `<` to `&lt`, is there anyway we can skip that formatting?

**Comment by kyoshino** [comment]
Yeah, FlareTail.js basically uses [`textContent`](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) instead of [`innerHTML`](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML) to sanitize the content, preventing potential XSS attacks. I guess some workaround like the `safe` flag could be implemented, but even without `<br>` tags in HTML, the content can be formatted with CSS [`white-space: pre-wrap`](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space) maybe?

**Comment by shinglyu** [comment]
Ah you are right! `white-space: pre-wrap` does the trick. I should really spend some time to read through a CSS tutorial :P. Do you suggest any good CSS material?

**Comment by kyoshino** [comment]
MDN is comprehensive but complicated. I want to see a better reference site too :P
